smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -230,32 +240,40 @@ def _on_iteration(self) -> None:
             return
 
         # perform default deserialization of the message envelope
-        request_bytes: bytes = self._task_queue.get()
-
-        request = deserialize_message(request_bytes, self._comm_channel_type)
+        # perform default deserialization of the message envelope
+        with self._task_queue.recvh(timeout=None) as recvh:
+            try:
+                request_bytes, _ = recvh.recv_bytes(timeout=None)
+            except fli.FLIEOT as exc:
+                return
+
+        request = deserialize_message(
+            request_bytes, self._comm_channel_type, self._device
+        )
         if not self._validate_request(request):
             return
 
         # # let the worker perform additional custom deserialization
         # request = self._worker.deserialize(request_bytes)
 
         fetch_model_result = self._worker.fetch_model(request, self._feature_store)
-        model_result = self._worker.load_model(request, fetch_model_result)
+        model_result = self._worker.load_model(
+            request, fetch_model_result, self._device
+        )
         fetch_input_result = self._worker.fetch_inputs(request, self._feature_store)
-        transformed_input = self._worker.transform_input(request, fetch_input_result)
-
-        # batch: t.Collection[_Datum] = transform_result.transformed_input
-        # if self._batch_size:
-        #     batch = self._worker.batch_requests(transform_result, self._batch_size)
+        transformed_input = self._worker.transform_input(
+            request, fetch_input_result, self._device
```

Comment:
let's avoid adding new parameters to each stage and instead put `device` into the request object?

Reply:
I think `device` was removed from the request object and we want workers and worker manager to be associated to a device type